### PR TITLE
Adding f2py signature file to avoid using ctypes

### PR DESCRIPTION
**Pull request by gregstarr**
This PR is to fix apexpy on windows 10. I added a f2py signature file so that you don't have to use ctypes to use the `feldg` function in magfld.f. I added the signature file to the install script and removed `_cfa` from apex.py. I got all the tests to pass on Windows 10 and almost all of them to pass in Windows Subsystem for Linux. The failing ones I hope will pass in CI.

### CODE REVIEW

**Comment by gregstarr** [comment]
Passed everything in 2.7-64 and 3.6-64, passed almost everything in 3.5-64. Couldn't compile for the others. Any ideas?

**Comment by aburrell** [comment]
I suspect (but do not know for sure) that the .pyf file needs to be created on each system, and the systems that are passing are those most related to your system.  I think you're on the right track with upgrading the ctypes calls in the Apex class though.

**Comment by gregstarr** [comment]
The same compilation / build errors occurred in the last commit so I don't think it has to do with fortranapex.pyf

https://ci.appveyor.com/project/aburrell/apexpy/builds/37593205

**Comment by aburrell** [comment]
Also it passes on Travis, so that makes it more unlikely to be the cause.  

AACGMV2 had an appveyor problem recently and basically, it was down to fixing the testing environment (C was only available on limited platforms).  Could be something similar.

**Comment by gregstarr** [comment]
Looking back, it doesn't look like the "32" builds ever passed. I see this error in most of them `cannot find -lpython36
collect2.exe: error: ld returned 1 exit status`

**Comment by gregstarr** [comment]
Took a lot of trial and error (sorry about all the commits), but I got both 64 and 32 bit jobs working, as well as the PEP8 checker job. There are three tests failing on 64 bit python 3.5 related to the command line interface. I removed testing for python 3.4 because it looks like it is no longer supported by anaconda and I added 3.7. 

build inspired by: https://github.com/timcera/wdmtoolbox/blob/master/appveyor.yml

job early cancel: https://github.com/ogrisel/python-appveyor-demo/blob/master/appveyor.yml

**Comment by aburrell** [comment]
I saw all the commits this morning and was cheering you on, thank you for figuring this out!  I propagated the changes you made to the appveyor template (the CI templates are located in `ci/templates` and are used along with `setup.cfg` to update `tox.ini`, `appveyor.yml`, and `.travis.yml`.  I also:
- added more recent python versions
- updated the appropriate docs.

Hopefully this will all pass and we can merge this in!

File: appveyor.yml [hunk 1]
```diff
@@ -61,28 +62,32 @@ environment:
...
+  # This quickly fails the job if there is another pipeline queued ahead of it
+  - ps: if ($env:APPVEYOR_PULL_REQUEST_NUMBER -and $env:APPVEYOR_BUILD_NUMBER -ne ((Invoke-RestMethod `
+          https://ci.appveyor.com/api/projects/$env:APPVEYOR_ACCOUNT_NAME/$env:APPVEYOR_PROJECT_SLUG/history?recordsNumber=50).builds | `
+          Where-Object pullRequestId -eq $env:APPVEYOR_PULL_REQUEST_NUMBER)[0].buildNumber) { `
+            throw "There are newer queued builds for this pull request, failing early." }
   - '%MINICONDA_HOME%\Scripts\conda config --set always_yes yes'
   - '%MINICONDA_HOME%\Scripts\conda update -q conda'
-  - '%MINICONDA_HOME%\Scripts\conda create -q -p C:\pythontest python=%PYTHON_VERSION% %INSTALL_LIBPYTHON% pytest numpy'
-  # remove file which conflicts with installation of package 'readme' on case-insensitive file systems
+  - '%MINICONDA_HOME%\Scripts\conda create -q -p C:\pythontest python=%PYTHON_VERSION% pytest numpy libpython'
```
Comment (aburrell) [inline appveyor.yml:72]: Do we still need `libpython`?

File: ci/templates/appveyor.yml [hunk 2]
```diff
@@ -37,28 +24,32 @@ environment:
...
+  # This quickly fails the job if there is another pipeline queued ahead of it
+  - ps: if ($env:APPVEYOR_PULL_REQUEST_NUMBER -and $env:APPVEYOR_BUILD_NUMBER -ne ((Invoke-RestMethod `
+          https://ci.appveyor.com/api/projects/$env:APPVEYOR_ACCOUNT_NAME/$env:APPVEYOR_PROJECT_SLUG/history?recordsNumber=50).builds | `
+          Where-Object pullRequestId -eq $env:APPVEYOR_PULL_REQUEST_NUMBER)[0].buildNumber) { `
+            throw "There are newer queued builds for this pull request, failing early." }
   - '%MINICONDA_HOME%\Scripts\conda config --set always_yes yes'
   - '%MINICONDA_HOME%\Scripts\conda update -q conda'
-  - '%MINICONDA_HOME%\Scripts\conda create -q -p C:\pythontest python=%PYTHON_VERSION% %INSTALL_LIBPYTHON% pytest numpy'
-  # remove file which conflicts with installation of package 'readme' on case-insensitive file systems
+  - '%MINICONDA_HOME%\Scripts\conda create -q -p C:\pythontest python=%PYTHON_VERSION% pytest numpy libpython'
```
Comment (aburrell) [inline ci/templates/appveyor.yml:34]: If not, don't forget to remove it here first/as well.

File: src/apexpy/apex.py [hunk 3]
```diff
@@ -13,13 +13,11 @@
 # below try..catch required for autodoc to work on readthedocs
 try:
     from . import fortranapex as fa
-except:
+except ImportError as e:
```
Comment (aburrell) [inline src/apexpy/apex.py:16]:
```suggestion
except ImportError as err:
```

File: src/apexpy/apex.py [hunk 4]
```diff
@@ -13,13 +13,11 @@
...
     print("ERROR: fortranapex module could not be imported, so apexpy probably"
           " won't work.  Make sure you have a gfortran compiler. Wheels "
           "installation assumes your compiler lives in /opt/local/bin")
-
-import ctypes
-
+    raise e
```
Comment (aburrell) [inline src/apexpy/apex.py:20]:
```suggestion
    raise err
```

Comment (aburrell) [reply src/apexpy/apex.py:20]: To avoid single letter variable names that are not iterator indices.

**Comment by gregstarr** [comment]
Thanks for taking a look. I notice that the same 64 bit python 3.5 tests are failing on master as well, so perhaps that could be tracked in a separate issue?

File: src/apexpy/apex.py [hunk 5]
```diff
@@ -76,15 +74,15 @@ def __init__(self, date=None, refh=0, datafile=None, fortranlib=None):
...
 
         if date is None:
             self.year = helpers.toYearFraction(dt.datetime.now())
         else:
             try:
                 # convert date/datetime object to decimal year
                 self.year = helpers.toYearFraction(date)
-            except:
+            except AttributeError:
                 # failed so date is probably int/float, use directly
```
Comment (aburrell) [inline src/apexpy/apex.py:86]:
```suggestion
                # Failed while finding datetime attribute, so
                # date is probably an int or float; use directly
```

Just clarifying why the AttributeError is raised.

**Comment by aburrell** [comment]
> Thanks for taking a look. I notice that the same 64 bit python 3.5 tests are failing on master as well, so perhaps that could be tracked in a separate issue?

It could, but numpy cut active support for python 3.5, so I don't think it's a problem worth spending a lot of energy on.  Current lowest supported python is 3.6: https://numpy.org/neps/nep-0029-deprecation_policy.html

File: appveyor.yml (hunk 1 above)
Comment (gregstarr) [reply appveyor.yml:72]: Yes I think we do, adding libpython back in was one of the last things I had to do to get it to build. Don't ask me why though

**Comment by aburrell** [comment]
The tests that are failing are tests that were fixed in the last pull request.  Any idea what's up?

**Comment by gregstarr** [comment]
These are new failures, in the past the travis CI was only run on 2.7, 3.4, 3.5 and 3.6. Both 2.7 and 3.6 were passing before and are still passing now but the new ones 3.7, 3.8 and 3.9 are failing one test. It could be because of a change made in 3.7 on linux which messes with the test. I notice the test is trying to catch warnings and verify that there is exactly 1: 
`assert len(wmsg) == 1`
It is failing because `len(wmsg)` is 2. I will add a print statement so we can see whether a length of 2 is reasonable.

**Comment by gregstarr** [comment]
There was a numpy deprecation warning which was causing 2 warnings instead of the expected 1. Should be all set now, let me know if there are any adjustments you want.

File: tests/test_Apex.py [hunk 6]
```diff
@@ -393,6 +393,8 @@ def test_geo2apex_undefined_warning():
         apex2000 = Apex(date=2000, refh=10000)
         ret = apex2000.geo2apex(0, 0, 0)
 
+    print(wmsg)
+    print([w.message for w in wmsg])
```
Comment (aburrell) [inline tests/test_Apex.py:397]:
```suggestion
```

**Review by aburrell — COMMENTED** [review]
Fantastic, thanks for tracking that down.  I should have thought about deprecation warnings... ah well.  Only suggestion is to remove your debugging statements.  Should be good to go after that!

**Review by aburrell — APPROVED** [review]

**Pull request merged**